edit/configmap.vue

```diff
@@ -1,70 +1,99 @@
+    CruResource,
     NameNsDescription,
     KeyValue,
-    Footer,
-    ResourceTabs,
+    Labels,
     Tab,
+    Tabbed
   },
 
   mixins: [CreateEditView],
+
+  data() {
+    if (!this?.value?.spec?.type && !this.value?.spec) {
+      this.$set(this.value, 'spec', {});
+    }
+
+    return {};
+  },
 };
 </script>
 
 <template>
-  <form>
-    <NameNsDescription
-      :value="value"
-      :mode="mode"
-      name-label="Name"
-      :register-before-hook="registerBeforeHook"
-    />
-
-    <div class="spacer"></div>
-
-    <div v-if="!isView || Object.keys(value.data||{}).length" class="row">
-      <KeyValue
-        key="data"
-        v-model="value.data"
+  <CruResource
+    :done-route="doneRoute"
+    :mode="mode"
+    :resource="value"
+    :subtypes="[]"
+    :validation-passed="true"
+    :errors="errors"
+    @error="e=>errors = e"
+    @finish="save"
+    @cancel="done"
+  >
+    <template #define>
+      <NameNsDescription
+        v-if="!isView"
+        :value="value"
         :mode="mode"
-        :title="t('configmapPage.data.title')"
-        :protip="t('configmapPage.data.protip')"
-        :initial-empty-row="true"
+        name-label="Name"
+        :register-before-hook="registerBeforeHook"
       />
-    </div>
 
-    <div class="spacer"></div>
+      <div class="spacer"></div>
 
-    <ResourceTabs v-model="value" :mode="mode">
-      <template #before>
-        <Tab :label="t('configmapPage.tabs.binaryData.label')" name="binary-data">
+      <Tabbed :side-tabs="true">
+        <Tab name="data" :label="t('configmap.tabs.data.label')" :weight="1">
           <KeyValue
-            key="binaryData"
-            v-model="value.binaryData"
-            :protip="false"
+            key="data"
+            v-model="value.data"
             :mode="mode"
-            :add-allowed="false"
-            :read-accept="'*'"
-            :read-multiple="true"
-            :value-binary="true"
-            :value-base64="true"
-            :value-can-be-empty="true"
-            :initial-empty-row="false"
+            :protip="t('configmapPage.data.protip')"
+            :initial-empty-row="true"
           />
         </Tab>
-      </template>
-    </ResourceTabs>
-
-    <Footer :mode="mode" :errors="errors" @save="save" @done="done" />
-  </form>
+        <Tab name="binary-data" :label="t('configmap.tabs.binaryData.label')" :weight="2">
+          <div v-if="!isView || Object.keys(value.data||{}).length" class="row">
+            <KeyValue
+              key="binaryData"
+              v-model="value.binaryData"
+              :protip="false"
+              :mode="mode"
+              :add-allowed="false"
+              :read-accept="'*'"
+              :read-multiple="true"
+              :value-binary="true"
+              :value-base64="true"
+              :value-can-be-empty="true"
+              :initial-empty-row="true"
+            />
+          </div>
+        </Tab>
+        <Tab
+          name="labels-and-annotations"
```

Comment:
While working on the namespace page I also realized this needs to be rendered conditionally.

v-if="!isView"